packages/web/src/components/play-bar/desktop/PlayBar.js

```diff
@@ -503,22 +508,50 @@ class PlayBar extends Component {
 const makeMapStateToProps = () => {
   const getCurrentQueueItem = makeGetCurrent()
 
-  const mapStateToProps = (state, props) => ({
-    accountUser: getAccountUser(state),
-    currentQueueItem: getCurrentQueueItem(state),
-    playCounter: getCounter(state),
-    collectible: getCollectible(state),
-    isPlaying: getPlaying(state),
-    isBuffering: getBuffering(state),
-    playingUid: getPlayingUid(state),
-    playbackRate: getPlaybackRate(state),
-    lineupHasTracks: getLineupHasTracks(
+  const mapStateToProps = (state) => {
+    const premiumTrackSignatureMap = getPremiumTrackSignatureMap(state)
+    const lineupEntries = getLineupEntries(
       getLineupSelectorForRoute(state),
       state
-    ),
-    userId: getUserId(state),
-    theme: getTheme(state)
-  })
+    )
+
+    // The lineup has accessibly tracks when there is at least one track
+    // the user has access to i.e. a non-gated track or an unlocked gated track.
+    // This helps us determine whether there is a playable track.
+    // For example, in the case of a gated track that the user does not have
+    // access to, the track card / tile will not be playable because the
+    // play button is disabled; however, a user can still attempt to play
+    // the track by pressing spacebar. This will prevent the track
+    // from attempting to play in that case, and consequently avoiding the
+    // infinite loading loop on the playbar.
+    const lineupHasAccessibleTracks = lineupEntries.some((entry) => {
+      if (entry.kind !== Kind.TRACKS) return false
+
+      const { id } = entry
+      const {
+        is_premium: isPremium,
+        premium_content_signature: premiumContentSignature
+      } = getTrack(state, { id }) ?? {}
+
+      const hasPremiumContentSignature =
+        !!premiumContentSignature || !!premiumTrackSignatureMap[id]
+      return !isPremium || hasPremiumContentSignature
```

Comment:
nit: also see this logic repeated a bunch of times, maybe useful to extract to separate function.

Reply:
yeah there was some work done here that was reverted. will re-consider refactoring this piece too when that work becomes un-reverted